pandas/core/arrays/period.py

```diff
@@ -497,7 +493,9 @@ def _time_shift(self, n, freq=None):
         freq : pandas.DateOffset, pandas.Timedelta, or string
             Frequency increment to shift by.
         """
-        values = self._data + n * self.freq.n
+        if freq is not None:
```

Comment:
can you elaborate on the TODO?

Reply:
fleshed this out a bit

Reply:
I still don't understand the comment. It says "don't silently ignore", but it raises an error so that is not silently?

But looking at the code, I don't see the argument used anywhere (except for in DatetimeLikeArray.shift, but that is overridden in PeriodArray.shift). 
So I would rather simply remove this here.

Reply:
I added the raising because I figured it is better than silently ignoring, but I do not know what the ideal behavior is.  @TomAugspurger I think you added this behavior recently, can you clarify?

Reply:
This looks fine as is. I would prefer that the signature matches between the subclasses. Raising if it's specified seems better than ignoring.

I notice now that the positional argument should be `periods`, not `n`. I can fix that after the.

Reply:
I guess the TODO can be removed, and the error message updated, and maybe changed to a TypeError.

Reply:
Should the docstring be updated?  I'm inclined to leave this for the next pass.

Reply:
Fine by me.

Reply:
I would just update/remove the comment here (since it is here that you add it, and the comment is not correct)

Reply:
Done (pushing shortly)